### PR TITLE
Ensure node-e2e-test can open enough files

### PR DESCRIPTION
**Pull request by odinuge**
#### What type of PR is this?

/kind failing-test

#### What this PR does / why we need it:

The apiserver and test runner in node e2e runs under the sshd daemon that can limit the
number of files it can open. Set a higher limit to address the issues.

Without this PR:

```
odin@test-cos-beta-89-16108-403-11 ~ $ sudo netstat -tulpn | grep 6443
tcp        0      0 127.0.0.1:6443          0.0.0.0:*               LISTEN      5024/e2e_node.test  
odin@test-cos-beta-89-16108-403-11 ~ $ cat /proc/5024/limits | grep files
Max open files            1024                 4096                 files     
```

With this PR

```
odin@test-cos-beta-89-16108-403-11 ~ $ sudo netstat -tulpn | grep 6443
tcp        0      0 127.0.0.1:6443          0.0.0.0:*               LISTEN      6113/e2e_node.test  
odin@test-cos-beta-89-16108-403-11 ~ $ cat /proc/6113/limits | grep files
Max open files            1000000              1000000              files  
```

#### Which issue(s) this PR fixes:

xref #102148
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by cynepco3hahue** [comment]
@odinuge Can you please verify it locally? IIRC we have only the periodic serial job, so it not really possible to verify the PR via CI before the merging.

File: test/e2e_node/services/apiserver.go
Comment (ehashman) [inline test/e2e_node/services/apiserver.go:94]: This matches the solution proposed in https://github.com/kubernetes/kubernetes/issues/102148#issuecomment-844866768

Although this could potentially be reducing the rlimit if it was set higher?

We should really test this locally if we don't have a presubmit job.

**Comment by odinuge** [comment]
Without this PR:

```
odin@test-cos-beta-89-16108-403-11 ~ $ sudo netstat -tulpn | grep 6443
tcp        0      0 127.0.0.1:6443          0.0.0.0:*               LISTEN      5024/e2e_node.test  
odin@test-cos-beta-89-16108-403-11 ~ $ cat /proc/5024/limits | grep files
Max open files            1024                 4096                 files     
```

With this PR

```
odin@test-cos-beta-89-16108-403-11 ~ $ sudo netstat -tulpn | grep 6443
tcp        0      0 127.0.0.1:6443          0.0.0.0:*               LISTEN      6113/e2e_node.test  
odin@test-cos-beta-89-16108-403-11 ~ $ cat /proc/6113/limits | grep files
Max open files            1000000              1000000              files  
```

Will cleanup the comment, and then we can get this merged.

File: test/e2e_node/services/apiserver.go
Comment (odinuge) [reply test/e2e_node/services/apiserver.go:94]: > Although this could potentially be reducing the rlimit if it was set higher?

This is the default used in kubelet. If the apiserver have more than `1000000` open files, I think we have other problems :sweat_smile: 

> We should really test this locally if we don't have a presubmit job.

See comment below.

**Comment by odinuge** [comment]
Hmm. No output at all from pull-kubernetes-node-kubelet-serial... Strange.

/test pull-kubernetes-node-kubelet-serial

**Comment by fromanirh** [comment]
/lgtm
for all the good reasons emerged during the investigation.
Great work @odinuge

**Review by mrunalp — APPROVED** [review]

**Comment by ehashman** [comment]
/milestone v1.22

Test-only change, need to unblock the Node serial job to help with https://github.com/kubernetes/kubernetes/issues/102676#issuecomment-880657160

**Pull request merged**
